CoursesApiHttpTrigger/tests.py

```diff
@@ -0,0 +1,371 @@
+
+        # Do some checking of the returned error message.
+        error_msg = json.loads(resp.get_body().decode('utf-8'))
+        print(error_msg)
+        self.assertEqual(error_msg['errors'][0]["error"], "Bad Request")
+        self.assertEqual(
+            error_msg['errors'][0]["error_values"][0]['Parameter Error'],
+            'Invalid parameter passed')
+
+    def test_endpoint_with_invalid_char_4_mode(self):
+
+        institution_id = '10000055'
+        course_id = 'AB37'
+        mode = '4'  # 4 is an invalid char for mode
+
+        # Construct a mock HTTP request.
+        req = func.HttpRequest(
+            method='GET',
+            body=None,
+            url=
+            f'/api/institutions/{institution_id}/courses/{course_id}/modes/{mode}',
+            params={'version': '1'},
+            route_params={
+                'institution_id': institution_id,
+                'course_id': course_id,
+                'mode': mode
+            })
+
+        # Call the function for the endpoint.
+        resp = main(req)
+
+        # Check status code
+        self.assertEqual(resp.status_code, 400)
+
+        # Check content type
+        headers = dict(resp.headers)
+        self.assertEqual(headers['content-type'], 'application/json')
+
+        # Do some checking of the returned error message.
+        error_msg = json.loads(resp.get_body().decode('utf-8'))
+        print(error_msg)
+        self.assertEqual(error_msg['errors'][0]["error"], "Bad Request")
+        self.assertEqual(
+            error_msg['errors'][0]["error_values"][0]['Parameter Error'],
+            'Invalid parameter passed')
+
+    def test_endpoint_with_invalid_char_semicolon_version(self):
+
+        institution_id = '10000055'
+        course_id = 'AB37'
+        mode = '1'
+        version = ';'  # semicolon is an invalid char for version
+
+        # Construct a mock HTTP request.
+        req = func.HttpRequest(
+            method='GET',
+            body=None,
+            url=
+            f'/api/institutions/{institution_id}/courses/{course_id}/modes/{mode}',
+            params={'version': version},
+            route_params={
+                'institution_id': institution_id,
+                'course_id': course_id,
+                'mode': mode
+            })
+
+        # Call the function for the endpoint.
+        resp = main(req)
+
+        # Check status code
+        self.assertEqual(resp.status_code, 400)
+
+        # Check content type
+        headers = dict(resp.headers)
+        self.assertEqual(headers['content-type'], 'application/json')
+
+        # Do some checking of the returned error message.
+        error_msg = json.loads(resp.get_body().decode('utf-8'))
+        print(error_msg)
+        self.assertEqual(error_msg['errors'][0]["error"], "Bad Request")
+        self.assertEqual(
+            error_msg['errors'][0]["error_values"][0]['Parameter Error'],
+            'Invalid parameter passed')
+
+    # TODO add more tests
```

Comment:
So for mode i would add test to fail when `mode` is set to a value of `one`

Reply:
Done.